db/compaction/compaction_service_test.cc

```diff
@@ -322,6 +329,43 @@ TEST_F(CompactionServiceTest, ManualCompaction) {
   VerifyTestData();
 }
 
+TEST_F(CompactionServiceTest, CancelCompactionOnRemoteSide) {
+  Options options = CurrentOptions();
+  options.disable_auto_compactions = true;
+  ReopenWithCompactionService(&options);
+  GenerateTestData();
+
+  auto my_cs = GetCompactionService();
+
+  std::string start_str = Key(15);
+  std::string end_str = Key(45);
+  Slice start(start_str);
+  Slice end(end_str);
+  uint64_t comp_num = my_cs->GetCompactionNum();
+
+  // Test cancel compaction at the beginning
+  my_cs->SetCanceled(true);
+  auto s = db_->CompactRange(CompactRangeOptions(), &start, &end);
+  ASSERT_TRUE(s.IsIncomplete());
+  // compaction number is not increased
+  ASSERT_GE(my_cs->GetCompactionNum(), comp_num);
+  VerifyTestData();
+
+  // Test cancel compaction in progress
+  my_cs->SetCanceled(false);  // reset
+  SyncPoint::GetInstance()->SetCallBack(
+      "CompactionJob::Run():Inprogress",
+      [&](void* /*arg*/) { my_cs->SetCanceled(true); });
+
+  s = db_->CompactRange(CompactRangeOptions(), &start, &end);
```

Comment:
I think this is coming from a saved background error actually. Maybe assert `my_cs` became true again after the `CompactRange()`.

Does it still work if you make this `CompactRange()` in a fresh DB or separate test case?

Reply:
Nice catch, I updated the test to reopen and then run `CompactRange()` again, as the compaction failure is a background failure which is not recoverable.